web/views.py

```diff
@@ -157,7 +157,7 @@ def get_form_kwargs(self):
         return kwargs
 
     def form_invalid(self, **kwargs):
-        return self.render_to_response(self.get_context_data(**kwargs))
+        return self.render(self.get_context_data(**kwargs))
```

Comment:
There's a lot of pathological/superfluous code in here because the view is not using the proper abstraction for a multi form page - [FormSets](https://docs.djangoproject.com/en/3.0/topics/forms/formsets/) are what we want here. This should be a simple enough refactor to include in this PR

Reply:
FormSets seem to be only for multiple forms of the same type. But this does seem closely analogous to the exp/StudyDetailView which has several (manually created) forms and handles them in a single POST function?
I'd rather let this be for now since I haven't actually touched it (except to change `render_to_response` to `render` per Django update) and refactor along with the StudyDetailView. I suspect it needs better test coverage anyway so may take a little more time.

Reply:
mostly copied over from slack:
You're totally right about FormSets - my thought was that you should use the formset factory to target different fields on the same model, but that's not really what the abstraction is designed for. The correct layer for the form-choosing logic is the view itself.

Basically, if we’re intent on maintaining Django’s separation between password change and user data update forms, then we should have password change handled in a separate view. Otherwise we should just create a new ModelForm that can handle password changes as well. Alternatively, we could override get_form or get_form_class and do it that way if we really want to keep the forms separate and in the same view.